recipes/hdf5/all/conanfile.py

```diff
@@ -72,6 +73,11 @@ def validate(self):
         if hasattr(self, "settings_build") and tools.cross_building(self, skip_x64_x86=True):
             # While building it runs some executables like H5detect
             raise ConanInvalidConfiguration("Current recipe doesn't support cross-building (yet)")
+        if self.options.parallel:
+            if self.options.enable_cxx:
+                raise ConanInvalidConfiguration("Parallel and C++ options are mutually exclusive")
+            if self.options.get_safe("threadsafe", False):
+                raise ConanInvalidConfiguration("Parallel and Threadsafe options are mutually exclusive")
```

Comment:
as this... parallel but not thread safe, wut?

Reply:
Yes it's a bit confusing. They refer to two different things in HDF5
* `threadsafe` is for parallel access in shared memory systems - it basically just serializes access using a mutex.
* `parallel` refers to the HDF5 Parallel library which allows true parallel access using process level parallelism via MPI, so no shared memory.